tests/e2e_test/module_deletion_test.go

```diff
@@ -97,21 +97,32 @@ var _ = Describe("Non Blocking Kyma Module Deletion", Ordered, func() {
 		})
 
 		It("Given the Kyma Module is under deletion", func() {
-			By("When the Module is re-enabled")
+			By("When the Module is re-enabled in different channel")
+			module.Channel = "fast"
 			Eventually(EnableModule).
 				WithContext(ctx).
 				WithArguments(runtimeClient, defaultRemoteKymaName, remoteNamespace, module).
 				Should(Succeed())
-			By("Then the Module Manifest CR is still in a \"Deleting\" State")
-			Consistently(CheckManifestIsInState).
+			By("Then the Module will be updated on SKR")
+			Eventually(DeploymentIsReady).
 				WithContext(ctx).
-				WithArguments(kyma.GetName(), kyma.GetNamespace(), module.Name, controlPlaneClient, shared.StateDeleting).
+				WithArguments("template-operator-v2-controller-manager", "template-operator-system", runtimeClient).
 				Should(Succeed())
+			By("And the old deployment will be removed")
+			Consistently(DeploymentIsReady).
+				WithContext(ctx).
+				WithArguments("template-operator-controller-manager", "template-operator-system", runtimeClient).
+				Should(Not(Succeed()))
 			By("And the Module Default CR is in a \"Deleting\" State")
 			Consistently(CheckSampleCRIsInState).
 				WithContext(ctx).
 				WithArguments("sample-yaml", "kyma-system", runtimeClient, "Deleting").
 				Should(Succeed())
+			By("And the Module Manifest CR is still in a \"Deleting\" State")
+			Consistently(CheckManifestIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), "template-operator", controlPlaneClient, shared.StateDeleting).
+				Should(Succeed())
 		})
 
 		It("When the blocking finalizers from the Default Module CR get removed", func() {
```

Comment:
here in this step, can you add two more checks to verify  Module Default CR is in `Ready` state, and `Consistently (DeploymentIsReady)`, just to be bit solid.